### PR TITLE
param fixes to make passing args to tasks and data builders easier

### PR DESCRIPTION
**Pull request by mvcrouse**
## Related PRs
This PR is not dependent on any other PR

## What this PR does / why we need it

It simplifies the process of adding your own custom arguments by allowing arguments to be specified as task-specific, databuilder-specific, or for the main process.

## If applicable**
- [X] this PR contains documentation
- [ ] this PR contains unit tests
- [X] this PR has been tested for backwards compatibility

### CODE REVIEW

File: fms_sdg/generate_data.py [hunk 1]
```diff
@@ -19,19 +19,14 @@
...
-    include_data_path: str,
-    include_builder_path: str,
-    prompt_file_path: str,
-    console_output: bool = True,
-    restart_generation: bool = False,
+    task_kwargs: Dict,
+    builder_kwargs: Dict,
+    include_data_path: Optional[str] = None,
+    include_builder_path: Optional[str] = None,
+    restart_generation: Optional[bool] = False,
```
Comment (drugilsberg) [inline fms_sdg/generate_data.py:29]: As we have a valid default why not?
```suggestion
    restart_generation: bool = False,
```

File: fms_sdg/databuilders/simple/generate.py [hunk 2]
```diff
@@ -32,7 +32,7 @@ def __init__(
         self,
         *args: Any,
         num_prompt_instructions: Optional[int] = 2,
-        prompt_file_path: Optional[str] = None,
+        prompt_file_path: Optional[str] = "prompt.txt",
         request_batch_size: Optional[int] = 5,
```
Comment (drugilsberg) [inline fms_sdg/databuilders/simple/generate.py:36]: As we have valid defaults why not?
```suggestion
        prompt_file_path: str = "prompt.txt",
        request_batch_size: int = 5,
```

File: fms_sdg/databuilders/simple/generate.py [hunk 3]
```diff
@@ -32,7 +32,7 @@ def __init__(
         self,
         *args: Any,
         num_prompt_instructions: Optional[int] = 2,
```
Comment (drugilsberg) [inline fms_sdg/databuilders/simple/generate.py:34]: As we have valid default, why not?
```suggestion
        num_prompt_instructions: int = 2,
```

**Review by drugilsberg — COMMENTED** [review]
Minor suggestions, looks very nice and improve flexibility, might open a PR to leverage those in NL2SQL pipelines.

File: fms_sdg/generate_data.py (hunk 1 above)
Comment (mvcrouse) [reply fms_sdg/generate_data.py:29]: Ah yes of course

**Review by sivasankalpp — APPROVED** [review]

**Pull request merged**